shadow.c

```diff
@@ -316,6 +364,48 @@ PHP_MINIT_FUNCTION(shadow)
  */
 PHP_MSHUTDOWN_FUNCTION(shadow)
 {
+	if (shadow_override_copy != NULL) {
+		/* Cleanup after our user-specified overrides. */
+		char *over = shadow_override_copy;
+		size_t over_len;
+		char c;
+		int argno;
+		int argtype;
+
+		while(*over) {
+			char *next = strchr(over, ',');
+			if(!next) {
+				next = over+strlen(over);
+			}
+			for(over_len=0;over_len<next-over;over_len++) {
+				/* find @ or , */
+				if(over[over_len] == '@' || over[over_len] == ',' || over[over_len] =='\0') break;
+			}
+			argno = 0;
+			argtype = 0;
+			if(over[over_len] == '@') {
+				if(!isdigit(over[over_len+1])) {
+					if(over[over_len+1] == 'w') {
+						argtype = 1;
+					}
+					argno = atoi(over+over_len+2);
+				} else {
+					argno = atoi(over+over_len+1);
+				}
+			}
+			c = over[over_len];
+			over[over_len] = '\0';
+			/* Undo the override here... */
+			shadow_undo_override(over, over_len, argno, argtype);
```

Comment:
I think it's a bit unnecessary, it should be enough just to free the memory. PHP_MSHUTDOWN_FUNCTION is called when process is being terminated

Reply:
I don't want to do that in case any subsequent shutdown handlers try to call a PHP internal function. I don't know that any do or will, but I'd rather put it back as it was before rather than risk a SEGV because we're being thrifty with code.